src/towncrier/test/test_builder.py

```diff
@@ -125,3 +125,10 @@ def test_orphan_with_dotted_number(self):
             parse_newfragment_basename("+orphan_12.3.feature", ["feature"]),
             ("+orphan_12.3", "feature", 0),
         )
+
+    def test_orphan_all_digits(self):
+        """Orphaned snippets can consist of only digits."""
+        self.assertEqual(
+            parse_newfragment_basename("+123.feature", ["feature"]),
+            ("+123", "feature", 0),
```

Comment:
Could you add another non-orphan test with an identifier looking like a commit sha? In the templates I've started using recently, I try to output PRs/issues, commits and arbitrary identifiers separately: https://github.com/cherrypy/cheroot/blob/3591a1c/docs/changelog-fragments.d/.towncrier-template.rst.j2#L59-L63.

It'd be nice if that remained functional. Hence an explicit test...

Reply:
I'd say that should be a separate PR -- that's an interesting functional test but not really tied to the changes being made here.

Reply:
The reason I brought it up is that I wasn't sure if this PR influences it. So it's kinda related as a possible regression. But I understand that it may be non-atomic and separate otherwise.

Reply:
Thanks for your feedback. 

This test is named `test_orphan_all_digits` . Based on that, I think that `+123` is ok.

I also think that using a git commit ref is also a valid test... but it should be a separate test.